crates/compiler/can/src/def.rs

```diff
@@ -2272,12 +2272,17 @@ fn canonicalize_pending_body<'a>(
 
     opt_loc_annotation: Option<Loc<crate::annotation::Annotation>>,
 ) -> DefOutput {
+    let loc_value = match &loc_expr.value {
+        ast::Expr::ParensAround(loc_value) => loc_value,
+        _ => &loc_expr.value,
+    };
```

Comment:
Probably makes sense to run a general canonicalization instead?

Reply:
I think it should work fine if you leave this file unchanged. The `match` below already has a `_ =>` branch which calls `canonicalize_expr`, and the rest of it is for sepcial-casing specific `Expr` variants, so I think you should be all set based on your changes to `expr.rs`! 😃

Reply:
Oh I just saw your comment in Zulip explaining why this is changed. Good catch!

One note about this code is that it will work for `foo = (\x -> x)` but not `foo = ((\x -> x))` because it only unwraps one level of parens.

What I'd recommend doing here is:
1. Remove this `let loc_value =`
2. Go back to `match (&loc_can_pattern.value, &loc_expr.value) {` below
3. Add a new branch to that `match` for `ast::Expr::ParensAround(loc_value) =>` which calls `canonicalize_pending_body` on that `loc_expr`

That way, no matter how many nested parens there are, it will unwrap all of them!

Reply:
Oh also I think it would be really great to add a test case for this (especially the double-wrapped-parens case) because I think it would be easy for someone to come along in the future, refactor this code somehow, and not realize their refactor broke this edge case. 😄

Reply:
1. I see, you suggest unwrapping the parens recursively. But isn’t it a fix for the special case then (my code is just a special case of the special case 😄)? My intuition is that the first step for the body canonicalization should be general canonicalization, and then the result is passed to the lambda-sets-magic-processor. But let me implement what you suggested first
2. Are there other places where we can have similar problems? I’ll look at all catchall match arms for `ast::Expr` values to check it.
3. Tests, sure!
4. The changes in this pr seem to be blowing up the stack. Are we moving in the right direction?